src/util/FetchUtil.ts

```diff
@@ -24,33 +42,37 @@ export async function fetchDataset(response: Response, converter: Representation
 Promise<Representation>;
 export async function fetchDataset(input: string | Response, converter: RepresentationConverter, body?: string):
 Promise<Representation> {
-  let response: Response;
   if (typeof input === 'string') {
```

Comment:
These 2 cases differ so drastically now that I would just create 2 different functions, one that takes as input a string, and one that takes as input a response, converter and optionally a body. The function is not used in that many places so should be easy to replace.

Reply:
Will split into `fetchDatsetFromUrl` and `fetchDatasetFromRespone`. This leads to some changes in other files, but most notably, some of those now have a useless `converter` argument in their constructor. May I remove those from the constructor and config files in this PR too?

Reply:
> Will split into `fetchDatsetFromUrl` and `fetchDatasetFromRespone`. 

Would suggest `fetchDataset` and `responseToDataset`

> May I remove those from the constructor and config files in this PR too?

That would be ideal yes.

Reply:
> You'll have to mock the rdf-dereference library now instead and probably do a few other changes there as well

Tried to mock the rdf-dereference lib but just can't get it to work. I've pushed all I have so far, will continue looking at it next week.